--- source:LL::set_ghost_data
+++ verus:LL::set_ghost_data
@@ -6,5 +6,17 @@
         Ghost(block_size): Ghost<nat>,
         Ghost(heap_id): Ghost<Option<HeapId>>,
     )
+        
     {
+        
+        self.data = Ghost(LLData {
+            fixed_page,
+            block_size,
+            page_id,
+            heap_id,
+            instance,
+            len: 0,
+            block_ids: Set::empty(),
+            idx_bound: 0,
+        });
     }
