--- source:struct_LL
+++ verus:struct_LL
@@ -1,8 +1,8 @@
 pub struct LL {
-    first: *mut Node,
+    pub(crate) first: *mut Node,
 
-    data: Ghost<LLData>,
+    pub(crate) data: Ghost<LLData>,
 
     // first to be popped off goes at the end
-    perms: Tracked<Map<nat, (PointsTo<Node>, PointsToRaw, Mim::block, IsExposed)>>,
+    pub(crate) perms: Tracked<Map<nat, (PointsTo<Node>, PointsToRaw, Mim::block, IsExposed)>>,
 }
