diff --git a/verus-mimalloc/layout.rs b/verus-mimalloc/layout.rs
--- a/verus-mimalloc/layout.rs
+++ b/verus-mimalloc/layout.rs
@@ -272,9 +272,10 @@ pub open spec fn calculate_inner_size(size: usize, capacity: usize) -> int {
 #[verifier::spinoff_prover]
 #[inline]
-#[verifier::external_body]
 pub fn align_down(x: usize, y: usize) -> (res: usize)
+    requires
+        y > 0,
     ensures
         y > 0 ==> (res as int) <= (x as int),
         y > 0 ==> (res as int) % (y as int) == 0,
         y > 0 ==> (x as int) < (res as int) + (y as int),
 {
@@ -290,7 +291,8 @@ pub fn align_down(x: usize, y: usize) -> (res: usize)
 }
 
 #[inline]
-#[verifier::external_body]
 pub fn align_up(x: usize, y: usize) -> (res: usize)
+    requires
+        y > 0,
     ensures
         y > 0 && (x as int) + (y as int) - 1 <= (usize::MAX as int) ==> (x as int) <= (res as int),
         y > 0 && (x as int) + (y as int) - 1 <= (usize::MAX as int) ==> (res as int) <= (x as int) + (y as int) - 1,
         y > 0 && (x as int) + (y as int) - 1 <= (usize::MAX as int) ==> (res as int) % (y as int) == 0,
